src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyRevealedMine.java

```diff
@@ -0,0 +1,301 @@
+
+            User buyer = database.getUser(dataDTO.userDTO.uuid);
+            Card boughtCard = database.getCard(dataDTO.cardDTO.uuid);
+            Room room = database.getRoomWithUser(buyer.getUuid());
+            Game game = room.getGame();
+
+            buyer.buyCard(boughtCard);
+            Card cardDrawn = game.takeCardFromRevealed(boughtCard);
+            
+            Log.DEBUG("User "+buyer.getName()+" has bought card ("+boughtCard.getUuid()+")");
+            Log.DEBUG("New card drawn "+cardDrawn.getUuid());
+
+            ResponseData responseData = new ResponseData(
+                new BuyerDataResponse(
+                    buyer.getUuid(), 
+                    new TokensDataResponse(
+                        buyer.getTokenCount(TokenType.RUBY), 
+                        buyer.getTokenCount(TokenType.EMERALD), 
+                        buyer.getTokenCount(TokenType.SAPPHIRE), 
+                        buyer.getTokenCount(TokenType.DIAMOND), 
+                        buyer.getTokenCount(TokenType.ONYX), 
+                        buyer.getTokenCount(TokenType.GOLD_JOKER)
+                    ), 
+                    boughtCard.getUuid()
+                ), 
+                new CardDataResponse(
+                    cardDrawn.getUuid(),
+                    cardDrawn.getCardTier(),
+                    cardDrawn.getAdditionalToken(),
+                    cardDrawn.getPoints(),
+                    cardDrawn.getCost(TokenType.RUBY),
+                    cardDrawn.getCost(TokenType.EMERALD),
+                    cardDrawn.getCost(TokenType.SAPPHIRE),
+                    cardDrawn.getCost(TokenType.DIAMOND),
+                    cardDrawn.getCost(TokenType.ONYX))
+            );
+
+            ArrayList<User> players = room.getAllUsers();
+            ServerMessage serverMessage = new ServerMessage(
+                userMessage.getContextId(), 
+                ServerMessageType.BUY_REVEALED_MINE_ANNOUNCEMENT, 
+                Result.OK, 
+                responseData
+            );
+
+            for(User player : players){
+                messenger.addMessageToSend(player.getConnectionHashCode(), serverMessage);        
+            }
+
+
+        }catch (Exception e){
+            ErrorResponse errorResponse = new ErrorResponse(
+                Result.FAILURE, 
+                e.getMessage(), 
+                ServerMessageType.BUY_REVEALED_MINE_RESPONSE, 
+                userMessage.getContextId().toString()
+            );
+            messenger.addMessageToSend(connectionHashCode, errorResponse);
+        }
+    }
+
+    private void validateData(DataDTO dataDTO,Database database) throws 
+    UserNotAMemberException, UserAlreadyInRoomException, RoomDoesntExistException, UserTurnException, CardDoesntExistException{
+        
+        User player = database.getUser(dataDTO.userDTO.uuid);
+        
+        //Check if user exits    
+        if(player == null){
+            //TO DO 
```

Comment:
It is one line implementation. If you check the room existance, make the same thing with the user

Reply:
It will be implemented when startGame is implemented beacuse of exception class included in that PR.